src/TpetraLinearSystem.C

```diff
@@ -155,28 +155,28 @@ int TpetraLinearSystem::getDofStatus(stk::mesh::Entity node)
   bool has_non_matching_boundary_face_alg = false;
   bool hasPeriodic = false;
 
-  bool debug = false;
+  if (NULL == realm_.allPeriodicInteractingSelector_ && realm_.allPeriodicInteractingParts_.size())
```

Comment:
I do not think that a selector needs to be a private data member as STK will do this for us behind the scenes. I think we simply need to store the partVector for each.